includes/lib/transifex-live-integration-rewrite.php

```diff
@@ -331,18 +331,19 @@ function home_url_hook( $url ) {
 	 * @param string $string The string to filter
 	 * @return string The filtered string
 	 */
-	 function the_content_hook( $string) {
- 		
- 		// Regular expression that extracts all urls from a string
- 		preg_match_all('#\b(https|http)?://[^,\s()<>]+(?:\([\w\d]+\)|([^,[:punct:]\s]|/))#', $string, $matchArray);
- 		// Iterate through all links, rewrite when needed
- 		foreach($matchArray as $match){
- 				if ( !Transifex_Live_Integration_Validators::is_hard_link_ok( $match[0] ) ) {
- 				    continue;
- 				}
- 		    $retlink = $this->reverse_hard_link( $this->lang, $match[0], $this->languages_map, $this->source_language, $this->rewrite_pattern );
- 				$string = str_replace($match[0], $retlink, $string);
- 		}
- 		return $string;
- 	}
+	function the_content_hook( $string) {
+	  // Regular expression that extracts all urls from a string
+	  $regexp = "<a\s[^>]*href=(\"??)([^\" >]*?)\\1[^>]*>(.*)<\/a>";
+	  preg_match_all("/$regexp/siU", $string, $matchArray);
+	  // Iterate through all links, rewrite when needed
+	  foreach($matchArray[2] as $match){
+	    if ( !Transifex_Live_Integration_Validators::is_hard_link_ok( $match ) ) {
+	        continue;
+	      }
+	    $retlink = $this->reverse_hard_link( $this->lang, $match, $this->languages_map, $this->source_language, $this->rewrite_pattern );
+	    $string = str_replace($match, $retlink, $string);
+		}
+		return $string;
+	}
+
```

Comment:
@Diontsoumas both the old and the new code have inconsistent indentation in Github's online viewer. I see that this files uses both spaces and tabs for indentation. I suggest using only tabs, since all of the project's files do so.

Reply:
I have converted spaces to tabs, it looks a bit weird, but hopefully it is aligned with the rest of the code?

Reply:
@Diontsoumas looks good now.